File: backend/src/Hrms.Domain/Entities/Employee.cs
using Hrms.Domain.Common;

namespace Hrms.Domain.Entities;

/// <summary>
/// ข้อมูลพนักงานหลัก (Employee Core Master)
/// แมปกับตาราง hrms.employee
/// </summary>
public class Employee : BaseEntity
{
    public string EmployeeCode { get; set; } = string.Empty;
    public string? BiometricId { get; set; }
    /// <summary>สถานะการจ้างงาน: ACTIVE, PROBATION, RESIGNED, INACTIVE</summary>
    public string EmploymentStatus { get; set; } = "ACTIVE";
    public string? Prefix { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;

    // ข้อมูล PDPA ที่มีความอ่อนไหวสูง (Sensitive Data)
    public string? CitizenId { get; set; }
    public byte[]? CitizenIdEncrypted { get; set; }
    public string? CitizenIdMasked { get; set; }

    public DateOnly? BirthDate { get; set; }
    public string? Gender { get; set; }
    public long? GenderId { get; set; }

    public string? Nationality { get; set; }
    public long? NationalityId { get; set; }

    public string? Religion { get; set; }
    public long? ReligionId { get; set; }

    public string? MaritalStatus { get; set; }
    public long? MaritalStatusId { get; set; }

    public string? MilitaryStatus { get; set; }
    public bool IsTopLevel { get; set; } = false;

    // ข้อมูลลดหย่อนภาษีครอบครัว (Family Tax Deductions)
    public bool SpouseHasIncome { get; set; } = false;
    public int NumberOfChildren { get; set; } = 0;
    public int ParentDeductionCount { get; set; } = 0;
    public int DisabilityDeductionCount { get; set; } = 0;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? AvatarUpdatedAt { get; set; }

    // Helper property สำหรับแสดงชื่อ-นามสกุลเต็ม
    public string FullName => $"{Prefix} {FirstName} {LastName}".Trim();

    // Navigation Properties
    public virtual EmployeeAvatar? Avatar { get; set; }
    public virtual UserAccount? UserAccount { get; set; }
    public virtual EmployeeContact? Contact { get; set; }
    public virtual EmployeeSocialSecurity? SocialSecurity { get; set; }
    public virtual ICollection<EmployeeAddress> Addresses { get; set; } = new List<EmployeeAddress>();
    public virtual ICollection<EmployeeBankAccount> BankAccounts { get; set; } = new List<EmployeeBankAccount>();
    public virtual ICollection<EmployeeEducation> Educations { get; set; } = new List<EmployeeEducation>();
    public virtual ICollection<FamilyMember> FamilyMembers { get; set; } = new List<FamilyMember>();
    public virtual ICollection<EmergencyContact> EmergencyContacts { get; set; } = new List<EmergencyContact>();
    public virtual ICollection<EmployeeAssignment> Assignments { get; set; } = new List<EmployeeAssignment>();
    public virtual ICollection<EmploymentContract> Contracts { get; set; } = new List<EmploymentContract>();
    public virtual ICollection<EmployeeStatusHistory> StatusHistories { get; set; } = new List<EmployeeStatusHistory>();
    public virtual ICollection<EmployeeSignature> Signatures { get; set; } = new List<EmployeeSignature>();
}

File: frontend/src/types/employee.ts
export interface EmployeeContact {
  personalPhone?: string;
  personalEmail?: string;
  organizationEmail?: string;
}

export interface EmployeeSocialSecurity {
  socialSecurityNoMasked?: string;
  hospitalName?: string;
  hospitalCode?: string;
}

export interface EmployeeAddress {
  id: number;
  addressType: string;
  addressLine?: string;
  subDistrict?: string;
  district?: string;
  province?: string;
  postalCode?: string;
  isCurrent: boolean;
}

export interface EmployeeBankAccount {
  id: number;
  bankId: number;
  bankCode?: string;
  bankName?: string;
  accountNumber: string;
  accountType?: string;
  accountName?: string;
  isPrimary: boolean;
  status: string;
}

export interface Employee {
  id: number;
  employeeCode: string;
  prefix?: string;
  firstName: string;
  lastName: string;
  fullName: string;
  citizenIdMasked?: string;
  birthDate?: string;
  gender?: string;
  genderId?: number;
  nationality?: string;
  nationalityId?: number;
  religion?: string;
  religionId?: number;
  maritalStatus?: string;
  maritalStatusId?: number;
  militaryStatus?: string;
  isTopLevel: boolean;
  spouseHasIncome: boolean;
  numberOfChildren: number;
  parentDeductionCount: number;
  disabilityDeductionCount: number;
  createdAt?: string;
  updatedAt: string;
  positionId?: number;
  positionName?: string;
  departmentName?: string;
  divisionName?: string;
  employeeType?: string;
  contact?: EmployeeContact;
  socialSecurity?: EmployeeSocialSecurity;
  addresses: EmployeeAddress[];
  bankAccounts: EmployeeBankAccount[];
  educations?: EmployeeEducation[];
  familyMembers?: FamilyMember[];
  emergencyContacts?: EmergencyContact[];
}

export interface FamilyMember {
  id?: number;
  relationshipType: string;
  prefix?: string;
  firstName: string;
  lastName?: string;
  citizenId?: string;
  citizenIdMasked?: string;
  birthDate?: string;
  maritalStatus?: string;
  educationStatus?: string;
}

export interface EmergencyContact {
  id?: number;
  relationship?: string;
  prefix?: string;
  firstName: string;
  lastName: string;
  address?: string;
  primaryPhone: string;
  secondaryPhone?: string;
  isPrimary?: boolean;
}

export interface EmployeeEducation {
  id?: number;
  educationLevel: string;
  institution: string;
  major?: string;
  graduationYear?: number;
  gpa?: number;
}

export interface CreateEmployeePayload {
  // ข้อมูลทั่วไป
  employeeCode: string;
  prefix?: string;
  firstName: string;
  lastName: string;
  citizenId?: string;
  gender?: string;
  genderId?: number;
  nationality?: string;
  religion?: string;

  // ข้อมูลส่วนบุคคล & ที่อยู่
  birthDate?: string;
  maritalStatus?: string;
  militaryStatus?: string;
  addressType?: string;
  addressLine?: string;
  subDistrict?: string;
  district?: string;
  province?: string;
  postalCode?: string;
  addresses?: Array<{
    addressType: string;
    addressLine?: string;
    subDistrict?: string;
    district?: string;
    province?: string;
    postalCode?: string;
    isCurrent?: boolean;
  }>;

  // การติดต่อ & การศึกษา
  personalEmail?: string;
  organizationEmail?: string;
  personalPhone?: string;
  educationLevel?: string;
  institution?: string;
  major?: string;
  graduationYear?: number;
  gpa?: number;

  // การเงิน & ตำแหน่งงาน
  bankName?: string;
  accountNumber?: string;
  positionId?: number;
  positionName?: string;
  employeeType?: string;

  // ครอบครัว & กรณีฉุกเฉิน
  familyMembers?: FamilyMember[];
  emergencyContact?: EmergencyContact;

  // อื่นๆ (เช่น ประกันสังคม)
  socialSecurityNo?: string;
  hospitalName?: string;
}
